tap_github/client.py

```diff
@@ -118,6 +122,14 @@ def raise_for_error(resp, source, stream, client, should_skip_404):
         # Don't raise a NotFoundException
         return None
 
+    if error_code == 422 and PAGINATION_EXCEED_MSG in response_json.get('message', ''):
```

Comment:
The error 422 can be raised for other reasons, that is why we are comparing the message to check if it's related to pagination.